File: [5 kyu] Beeramid/main.cpp
// Returns number of complete beeramid levels
int beeramid(int bonus, double price) {
    if (bonus <= 0.0) {
        return 0;
    }

    int total = 0;
    int level = 0;

    while (total <= bonus) {
        level++;
        total += price * level * level;
    }

    return level - 1;
}

File: [5 kyu] Beeramid/main.rs
fn beeramid (bonus: i32, price: f32) -> usize {
    if bonus <= 0 {
        return 0;
    }

    let mut total = 0.0;
    let mut level = 0;

    while total <= bonus as f32 {
        level += 1;
        total += price * level as f32 * level as f32;
    }

    level - 1
}